src/trapfunc.cpp

```diff
@@ -86,6 +86,33 @@ void trapfunc::bubble( Creature *c, const tripoint &p )
     g->m.remove_trap( p );
 }
 
+void trapfunc::glass( Creature *c, const tripoint &p )
+{
+    // tiny animals don't trigger glass trap
+    if( ( c != nullptr && c->get_size() == MS_TINY ) || c->is_hallucination() ) {
+        return;
+    }
+    if( c != nullptr ) {
+        c->add_msg_player_or_npc( m_warning, _( "You step on some glass!" ),
+                                  _( "<npcname> steps on some glass!" ) );
+        c->add_memorial_log( pgettext( "memorial_male", "Stepped on glass." ),
+                             pgettext( "memorial_female", "Stepped on glass." ) );
+        
+        monster *z = dynamic_cast<monster *>( c );
+        const char dmg = std::max( 0, rng( -10, 10 ) );
+        if( z != nullptr && dmg > 0 ) {
+            z->moves -= 80;
```

Comment:
So if it's the player or npc step on a trap, they don't lose movement, only monsters do?

Reply:
I mirrored the effect of nailed board which does the same for some reason (minus chance for tetanus). True, it's quite strange, so I may fix it in both instances if you want me to?

Reply:
I think that would make sense.